src/lib/fetchingUtils.js

```diff
@@ -49,6 +49,18 @@ function generateQueryUrl(serverUrl, queryObj) {
       case "id":
         urlStr += `${seperator}_id=${queryObj[param]}`;
         break;
+      case "identifier":
+        if (queryObj.system) {
+          urlStr += `${seperator}identifier=${queryObj.system}%7C${queryObj[param]}`;
+        } else {
+          urlStr += `${seperator}identifier=${queryObj[param]}`;
+        }
+        break;
+      case "system":
+        if (!queryObj.identifier) {
+          urlStr += `${seperator}identifier=${queryObj[param]}%7C`;
+        }
+        break;
```

Comment:
I have encountered a super fun bug though, introduced by a lack of canonical ordering in our query param iterating here. In short: we might generate a malformed search URL if user's provide a system query parameter first, and an identifier second. In this scenario, we never append content onto our `urlStr` starting with a `?`, since we would have done that with our system parameter but don't because an identifier exists. 

To replicate: 
- Create a new patient query
- Add `http://hospital.smarthealthit.org` as our system as the first search parameter
- Add anything as an identifier
- Click the search button
- Inspect the 404 in the Network pane, and observe the request URL has a chain of query params starting with `&`. 

Naive solution: We could have a single method for appending identifier/system information onto the urlStr that only changes the content if that which we are appending isn't already on the urlStr (urlStr.findIndex(newContent) === -1 or something). We then call this method in both system and identifier cases. 

Open to other solutions, but wanted to make sure I thought of something before pointing out this bug.